cmd/list.go

```diff
@@ -0,0 +1,44 @@
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"os"
+
+	"github.com/codegangsta/cli"
+	"github.com/exercism/cli/api"
+	"github.com/exercism/cli/config"
+)
+
+const msgExplainFetch = "In order to fetch a specific assignment, call the fetch command with a specific assignment.\n\nexercism fetch ruby/matrix"
+
+// List returns the full list of assignments for a given language
+func List(ctx *cli.Context) {
+	c, err := config.Read(ctx.GlobalString("config"))
+	if err != nil {
+		log.Fatal(err)
+	}
+	args := ctx.Args()
+
+	if len(args) != 1 {
+		msg := "Usage: execism list LANGUAGE"
+		log.Fatal(msg)
+	}
+
+	language := args[0]
+
+	problems, err := api.List(language, c.XAPI)
+	if err != nil {
+		log.Fatal(err)
+	}
+	GenerateOutput(problems, os.Stdout)
+}
+
+// Output writes out the list of assignments
+func GenerateOutput(problems []string, w io.Writer) {
```

Comment:
I don't really like exposing a function like `GenerateOutput` to the entire `cmd` package. What do you think of adding a new package that can hold on to track-related things? Earlier I've isolated some things into the `Homework` type in the `user` package. Would that approach work?